### ISSUE TITLE
<? Operator behaves weirdly while pattern matching 

### ISSUE THREAD

**Issue by yuzudev** [DanielXMoore/Civet#573]
```ts
switch value
 <? "string"
  value
 []
  value.map(&.toString())
 true
  value

// ↓↓↓ Civet 0.6.11
if (value instanceof "string") {
  return value;
} else if (Array.isArray(value) && value.length === 0) {
  return value.map(($) => $.toString());
} else if (value === true) {
  return value;
}

```

**Comment by edemaine** [comment]
It looks like we need to call `processBinaryOpExpression`. Other complex operations also don't work. For example:

```coffee
switch value
  == 5 < 10
    value
```

compiles to

```js
if (value == 5 < 10) {
  value
}
```

**Comment by edemaine** [comment]
Fixed in 0.6.12!
